File: NEW/TwoPointer/HappyNumber.cpp
using namespace std;

#include <iostream>
#include <vector>
#include <unordered_map>

int findSqSum(int num){
    int sum = 0, i;
    while(num>0){
        i = num%10;
        num = num/10;
        sum += i*i;
    }

    return sum;
}
bool findHappyNumber(const int num) {
    int sp= num;
    int fp= num;

    do{
        sp = findSqSum(sp);
        fp = findSqSum(findSqSum(fp));
        cout<<sp<<"--"<<fp<<endl;
    }while(sp!=fp);

    return sp==1;
} 

int main(int argc, char* argv[]) {
  bool result;
  result = findHappyNumber(23);
  cout << "Is Happy Number: " << result << endl;
  result = findHappyNumber(12);
  cout << "Is Happy Number: " << result << endl;
}

File: Python/fastAndSlowPointer/HappyNumber.py
def func(num):

    slow, fast = num, num
    while(True):
        slow = findSquareSum(slow)
        fast = findSquareSum(findSquareSum(fast))
        if(fast == slow):
            return slow == 1


def findSquareSum(num):
    _sum = 0
    while(num>0):
        _sum += (num%10)*(num%10)
        num = num//10
    return _sum

print(str(func(23)))#True, sum of squares is 1
print(str(func(12)))#False, not happy number